src/v/datalake/coordinator/state_update.cc

```diff
@@ -45,12 +47,22 @@ add_files_update::can_apply(const topics_state& state) {
     if (entries.empty()) {
         return stm_update_error{"No entries requested"};
     }
-    auto prt_state_opt = state.partition_state(tp);
-    if (!prt_state_opt.has_value()) {
-        // No entries at all, this partition hasn't ever added any files.
+    auto topic_it = state.topic_to_state.find(tp.topic);
+    if (topic_it == state.topic_to_state.end()) {
+        return std::nullopt;
+    }
+    auto cur_topic_rev = topic_it->second.revision;
+    if (topic_revision != cur_topic_rev) {
         return std::nullopt;
```

Comment:
> Can you add a comment explaining why this add operation should be allowed in the stm? It's seems reasonable to think the operation shouldn't be allowed

If we end up here, this means the previous revision has been fully purged and we are ready to open a new revision (because now addition of new topic states are still initiated by translators). Will add a comment

> Would be great to add some unit tests for these

Will do